backend/src/main/kotlin/app/ehrenamtskarte/backend/userdata/webservice/UserImportHandler.kt

```diff
@@ -0,0 +1,120 @@
+import org.apache.commons.csv.CSVParser
+import org.jetbrains.exposed.sql.select
+import org.jetbrains.exposed.sql.transactions.transaction
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import java.io.BufferedReader
+import java.io.InputStreamReader
+import java.time.LocalDate
+import java.time.format.DateTimeFormatter
+import java.time.format.DateTimeParseException
+
+class UserImportHandler {
+
+    private val logger: Logger = LoggerFactory.getLogger(UserImportHandler::class.java)
+
+    fun handle(ctx: Context) {
+        try {
+            // TODO as part of #1417:
+            // validate auth token, check to which project it belongs, check if this project is allowed to send user data
+            // (should be added as a flag in config.yml), save projectId into val for the later use, else return an error
+            val files = ctx.uploadedFiles("file")
+            when {
+                files.isEmpty() -> throw UserImportException("No file uploaded")
+                files.size > 1 -> throw UserImportException("Multiple files uploaded")
+            }
+            val file = files[0]
+
+            BufferedReader(InputStreamReader(file.content())).use { reader ->
+                getCSVParser(reader).use { csvParser ->
+                    importData(csvParser)
+                }
+            }
+            ctx.status(200).result("Import successfully completed")
+        } catch (exception: UserImportException) {
+            ctx.status(400).result(exception.message ?: "Import failed")
+        } catch (exception: Exception) {
+            logger.error("Failed to perform user import", exception)
+            ctx.status(500).result("Internal error occurred")
+        }
+    }
+
+    private fun getCSVParser(reader: BufferedReader): CSVParser {
+        return CSVParser(
+            reader,
+            CSVFormat.DEFAULT.builder()
+                .setHeader()
+                .setTrim(true)
+                .setSkipHeaderRecord(true)
+                .build()
+        )
+    }
+
+    private fun importData(csvParser: CSVParser) {
+        val headers = csvParser.headerMap.keys
+        val requiredColumns = setOf("regionId", "userHash", "startDate", "endDate", "revoked")
+        if (!headers.containsAll(requiredColumns)) {
+            throw UserImportException("Missing required columns: ${requiredColumns - headers}")
+        }
+
+        transaction {
+            // TODO as part of #1417: define the project by auth token
+            val project = ProjectEntity.find { Projects.project eq "bayern.ehrenamtskarte.app" }.single()
+            val regionsByProject = Regions.select { Regions.projectId eq project.id }.map { it[Regions.id].value }
+
+            for (entry in csvParser) {
+                if (entry.toMap().size != csvParser.headerMap.size) {
+                    throw UserImportException(entry.recordNumber, "Missing data")
+                }
+
+                val regionId = entry.get("regionId").toInt()
```

Comment:
i'm just not pretty sure if we should use sth not project based and use a lookup instead. Maybe at some point f.e. database migration from baselane or sth with less or more other regions from the bavaria project is needed and then we don't have the same regionId for koblenz.
Maybe we could use the `regionIdentifier` to get the `regionId` since it won't change @seluianova

Reply:
Just to make sure:
If for some reason we get the same userHash for different regions - 2 records will be created. Because there is a unique constraint - userHash + regionKey.
Do you think it's okay?

Reply:
Just to make sure:
If for some reason we get the same userHash for different regions - a record for each region will be created. Because I added a unique constraint: userHash + regionKey.
Do you think it's okay?

Reply:
Changed regionId to regionKey in the UserEntitlements table

Reply:
As discussed, I changed regionId to regionKey in the UserEntitlements table

And just to make sure:
If for some reason we get the same userHash for different regions - a record for each region will be created. Because I added a unique constraint: userHash + regionKey.
Do you see any issues with that?